From 6a547c27c5d359af376fdb277ea05eea0d76e870 Mon Sep 17 00:00:00 2001
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Fri, 30 Aug 2013 08:51:25 +0200
Subject: [PATCH 06/29] Remove no-hpet option

RH-Author: Miroslav Rezanina <mrezanin@redhat.com>
Message-id: <9d2bbee5227f35ee23da66a8689a9cd0faf545f1.1377851454.git.mrezanin@redhat.com>
Patchwork-id: 53936
O-Subject: [RHEL7 qemu-kvm PATCHv2 5/6] Remove no-hpet option
Bugzilla: 947441
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Michal Novotny <minovotn@redhat.com>

From: Miroslav Rezanina <mrezanin@redhat.com>

Bugzilla: 947441
Upstream: n/a, RHEL only

We disable HPET for RHEL, this patch removes option no-hpet that is
not usable.

Patch is forwardport of RHEL6 patch 1c07ac51 with small difference in
qemu-options.hx STEXI part of no-hpet is removed to keep man page
updated.

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>

v2:
 - Remove only STEXI part
---
 qemu-options.hx | 7 ++-----
 vl.c            | 4 +++-
 2 files changed, 5 insertions(+), 6 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 qemu-options.hx |    7 ++-----
 vl.c            |    4 +++-
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/qemu-options.hx b/qemu-options.hx
index 2ad0ed2..202a2cf 100644
--- a/qemu-options.hx
+++ b/qemu-options.hx
@@ -1283,13 +1283,10 @@ it if your guest OS complains about ACPI problems (PC target machine
 only).
 ETEXI
 
+#if 0 /* Disabled for Red Hat Enterprise Linux */
 DEF("no-hpet", 0, QEMU_OPTION_no_hpet,
     "-no-hpet        disable HPET\n", QEMU_ARCH_I386)
-STEXI
-@item -no-hpet
-@findex -no-hpet
-Disable HPET support.
-ETEXI
+#endif
 
 DEF("acpitable", HAS_ARG, QEMU_OPTION_acpitable,
     "-acpitable [sig=str][,rev=n][,oem_id=str][,oem_table_id=str][,oem_rev=n][,asl_compiler_id=str][,asl_compiler_rev=n][,{data|file}=file1[:file2]...]\n"
diff --git a/vl.c b/vl.c
index f7b326a..24536b5 100644
--- a/vl.c
+++ b/vl.c
@@ -217,7 +217,7 @@ int smp_threads = 1;
 const char *vnc_display;
 #endif
 int acpi_enabled = 1;
-int no_hpet = 0;
+int no_hpet = 1; /* Always disabled for Red Hat Enterprise Linux */
 int fd_bootchk = 1;
 static int no_reboot;
 int no_shutdown = 0;
@@ -3682,9 +3682,11 @@ int main(int argc, char **argv, char **envp)
             case QEMU_OPTION_no_acpi:
                 acpi_enabled = 0;
                 break;
+#if 0 /* Disabled for Red Hat Enterprise Linux */
             case QEMU_OPTION_no_hpet:
                 no_hpet = 1;
                 break;
+#endif
             case QEMU_OPTION_balloon:
                 if (balloon_parse(optarg) < 0) {
                     fprintf(stderr, "Unknown -balloon argument %s\n", optarg);
-- 
1.7.1

